### PR TITLE
More necromorphs

### PR DESCRIPTION
**Pull request by ghost**
Adds more necromorphs, as well as move the regular necromorph from monsters.dm to its own file.

>Leapers

leaper

They work basically like the frogs from the Tomb away mission, they run at people and latch onto them, causing severe damage.

The problem is, they also like to hide in vents, so care should be taken if you're going into necromorph territory.

>Exploder

exploder

A kamikazi unit, runs at its enemies and hits them with its abnormal growth on its left arm, exploding. It also explodes on death so watch your fire.

>Puker

puker

Must have been a chemist in its last life. When it sees an enemy, it fires projectile puke at it, which spreads polyacid, sulphuric acid, and vomit as it flies until it hits its target. This takes from the pukes internal reservoir, so there is a nice bit of damage dropoff.

Sprites courtesy of @Blithering 

:cl:

* rscadd: Adds more necromorph types

### CODE REVIEW

**Comment by ghost** [comment]
Can you post images?

**Comment by ghost** [comment]
Sure, hang on

**Comment by ghost** [comment]
Pictures added

**Comment by Ephx1** [comment]
You should add those horrible wall screaming necromorphs as a sort of turret with their little shooting spawn.

**Comment by ghost** [comment]
3 at a time, and the wall screaming necromorphs are a bit too stationary, considering there's no marker to defend just yet.

**Comment by Riekland** [comment]
Make sure it plays the screaming sound at 200% volume every tick with no way to stop it aside from killing it

**Comment by ghost** [comment]
Sound and spess station 13 don't go well together

**Comment by SkowronX** [comment]
Skin parts need more details and better transition to the bloody parts

**Comment by ghost** [comment]
add the cute baby necromorphs in the next batch

**Comment by Ephx1** [comment]
Also make the exploder's arm thing look less like a chicken wing.
Shame you won't have their sounds, the exploder's confused "EEEYAH" is great.

File: code/modules/mob/living/simple_animal/hostile/necromorph.dm [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	if(throwing && isliving(A) && CanAttack(A))
+		attach(A)
+	..()
+
+/mob/living/simple_animal/hostile/necromorph/leaper/Life()
+
+	update_climb()
+	if(!isUnconscious())
+		if(stance == HOSTILE_STANCE_IDLE && !client)
+			var/list/can_see = view(get_turf(src), vision_range/1.5) //Nothing too close for comfort
```
Comment (Probe1) [inline code/modules/mob/living/simple_animal/hostile/necromorph.dm]: This is still pretty far away.

File: code/modules/mob/living/simple_animal/hostile/necromorph.dm [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	..()
+	splash_sub(reagents, atarget, -1)
+
+/obj/item/projectile/puke/process_step()
+	..()
+	var/turf/simulated/T = get_turf(src)
+	if(T) //The first time it runs, it won't work, it'll runtime
+		playsound(T, 'sound/effects/splat.ogg', 50, 1)
+		T.add_vomit_floor(src, 1, 1, 1)
+	sleep(2) //Slow the fuck down, hyperspeed vomit
```
Comment (Probe1) [inline code/modules/mob/living/simple_animal/hostile/necromorph.dm]: Vomit is pretty easily dodgeable currently.  Maybe sleep(1)

File: code/modules/mob/living/simple_animal/hostile/necromorph.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/necromorph.dm]: Divided by 2 sound good? Then it goes from 9 to 6 to 4.5, which I'd assume would round up to 5?

File: code/modules/mob/living/simple_animal/hostile/necromorph.dm (hunk 2 above)
Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/necromorph.dm]: It's meant to be dodgeable, but if you insist

Comment (Probe1) [reply code/modules/mob/living/simple_animal/hostile/necromorph.dm]: Its _extremely_ dodgeable though.

File: code/modules/mob/living/simple_animal/hostile/necromorph.dm (hunk 1 above)
Comment (Probe1) [reply code/modules/mob/living/simple_animal/hostile/necromorph.dm]: Yeah 2 should be fine.  I'm not sure if it makes a difference.  Still, the range they detect you and leave the vent is 3-4 tiles which is way way more than enough for the player to say nope

**Comment by ghost** [comment]
I'm not sure what more should be done with this besides sprites

**Comment by Probe1** [comment]
Do you have a plan for the sprites or is it good enough as is

**Review by Probe1 — APPROVED** [review]

**Comment by ghost** [comment]
I'd have to ask blithering, he did them on request

**Comment by ghost** [comment]
woo

Wonder how these mobs'll break the server

**Pull request merged**